cider/function/FunctionLookup.cpp

```diff
@@ -0,0 +1,198 @@
+    scalar_function_look_up_ptr_ =
+        std::make_shared<cider::function::substrait::SubstraitScalarFunctionLookup>(
+            cider_internal_function_ptr, substrait_mappings);
+    aggregate_function_look_up_ptr_ =
+        std::make_shared<cider::function::substrait::SubstraitAggregateFunctionLookup>(
+            cider_internal_function_ptr, substrait_mappings);
+    extension_function_look_up_ptr_ =
+        std::make_shared<cider::function::substrait::SubstraitScalarFunctionLookup>(
+            substrait_extension_function_ptr, substrait_mappings);
+  } else if (from_platform == PRESTO_ENGINE) {
+    cider::function::substrait::SubstraitExtensionPtr presto_extension_function_ptr =
+        cider::function::substrait::SubstraitExtension::loadExtension(
+            {getDataPath() + "/presto/" + "presto_extension.yaml"});
+    cider::function::substrait::SubstraitFunctionMappingsPtr presto_mappings =
+        std::make_shared<
+            const cider::function::substrait::VeloxToSubstraitFunctionMappings>();
+    scalar_function_look_up_ptr_ =
+        std::make_shared<cider::function::substrait::SubstraitScalarFunctionLookup>(
+            cider_internal_function_ptr, presto_mappings);
+    aggregate_function_look_up_ptr_ =
+        std::make_shared<cider::function::substrait::SubstraitAggregateFunctionLookup>(
+            cider_internal_function_ptr, presto_mappings);
+    extension_function_look_up_ptr_ =
+        std::make_shared<cider::function::substrait::SubstraitScalarFunctionLookup>(
+            presto_extension_function_ptr, presto_mappings);
+  } else {
+    CIDER_THROW(CiderCompileException,
+                std::string("Function lookup unsupported platform ") + from_platform);
+  }
+}
+
+const SQLOpsPtr FunctionLookup::getFunctionScalarOp(
+    const FunctionSignature& function_signature) const {
+  const std::string& from_platform = function_signature.from_platform;
+  if (from_platform != from_platform_) {
+    CIDER_THROW(CiderCompileException,
+                std::string("Function lookup unsupported platform ") + from_platform);
+  }
+  const std::string& func_name = function_signature.func_name;
+  const auto& functionSignature =
+      cider::function::substrait::SubstraitFunctionSignature::of(
+          func_name, function_signature.arguments, function_signature.return_type);
+  const auto& functionOption =
+      scalar_function_look_up_ptr_->lookupFunction(functionSignature);
+  if (functionOption.has_value()) {
+    return function_mappings_->getFunctionScalarOp(func_name);
+  }
+  return nullptr;
+}
+
+const SQLAggPtr FunctionLookup::getFunctionAggOp(
+    const FunctionSignature& function_signature) const {
+  const std::string& from_platform = function_signature.from_platform;
+  if (from_platform != from_platform_) {
+    CIDER_THROW(CiderCompileException,
+                std::string("Function lookup unsupported platform ") + from_platform);
```

Comment:
`platform of target function is mismatched with registered platform.`

Reply:
Done.